src/ruleconfig.jl

```diff
@@ -0,0 +1,19 @@
+"""
+    ReverseRuleConfigBackend
+
+AD backend that uses reverse mode with any ChainRules-compatible reverse-mode AD package.
+"""
+struct ReverseRuleConfigBackend{RC <: RuleConfig} <: AbstractReverseMode
+    ruleconfig::RC
+end
+
+AD.@primitive function pullback_function(ab::ReverseRuleConfigBackend, f, xs...)
+    return (vs) -> begin
+        _, back = rrule_via_ad(ab.ruleconfig, f, xs...)
+        if vs isa Tuple && length(vs) === 1
+            return Base.tail(back(vs[1]))
```

Comment:
It seems this rules out support for functors `f` as it removes the derivatives wrt to `f`? But maybe generally the design of AbstractDifferentiation does not support them?

Reply:
Yes, this is so. AbstractDifferentiation supports a strictly smaller set of applications than, say, Zygote. It's all about scalars and arrays. And even there, I'm pretty sure results will be inconsistent if one tries to, say, take the gradient of a structured array. And functors are not supported.